From 6b52df10f04e4ed5b978ea86256124a85cd06e55 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 5 Dec 2012 13:41:45 +0800
Subject: [PATCH 380/635] ENGR00235672 Wifi: fix having same mac address issue

This issue already fix in ICS, On JB, missing to change the file path in
check_wifi_mac.sh which will cause getting mac from softmac failed for
permission error. Also fix one warning info which is caused by not used
variable.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 wlan/atheros/compat-wireless/check_wifi_mac.sh                        | 4 ++--
 .../atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c | 4 +---
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/wlan/atheros/compat-wireless/check_wifi_mac.sh b/wlan/atheros/compat-wireless/check_wifi_mac.sh
index 7b6e932..a47ad20 100755
--- a/wlan/atheros/compat-wireless/check_wifi_mac.sh
+++ b/wlan/atheros/compat-wireless/check_wifi_mac.sh
@@ -7,7 +7,7 @@ else
 	touch /device/wifi/softmac
 	echo 02:30:`busybox expr $RANDOM % 89 + 10`:`busybox expr $RANDOM % 89 + 10`:`busybox expr $RANDOM % 89 + 10`:`busybox expr $RANDOM % 89 + 10` > /device/wifi/softmac
 	sync
-	chmod 775 /vendor/wifi
-	chmod 664 /vendor/wifi/softmac
+	chmod 775 /device/wifi
+	chmod 664 /device/wifi/softmac
 	mount -r -o remount /device
 fi
diff --git a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c
index 41f08b7..f09ee66 100644
--- a/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c
+++ b/wlan/atheros/compat-wireless/drivers/net/wireless/ath/ath6kl/softmac.c
@@ -1,7 +1,6 @@
 /*
  * Copyright (c) 2011 Atheros Communications Inc.
- *
- * Copyright (c) 2010-2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012 Freescale Semiconductor, Inc.
  *
  * Permission to use, copy, modify, and/or distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -69,7 +68,6 @@ static void ath6kl_calculate_crc(u32 target_type, u8 *data, size_t len)
 
 static int ath6kl_fetch_mac_from_file(struct ath6kl *ar, u8* filename)
 {
-	char softmac_filename[256];
 	int ret = 0;
 
 	do {
-- 
1.8.0

